Tests/FluentAssertions.Specs/Primitives/HttpResponseMessageAssertionSpecs.cs

```diff
@@ -0,0 +1,442 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs.Primitives
+{
+    public class HttpResponseMessageAssertionSpecs
+    {
+        [Theory]
+        [InlineData(HttpStatusCode.OK)]
+        [InlineData(HttpStatusCode.Accepted)]
+        public void Should_succeed_when_status_code_is_successful(HttpStatusCode statusCodeOfResponse)
+        {
+            // Arrange
+            var testee = new HttpResponseMessage(statusCodeOfResponse);
+
+            // Act / Assert
+            testee.Should().BeSuccessful();
+        }
+
+        [Fact]
+        public void Should_fail_when_status_code_error_is_successful()
+        {
+            // Arrange
+            var testee = new HttpResponseMessage(HttpStatusCode.Gone);
+
+            // Act
+            Action action = () => testee.Should().BeSuccessful();
+
+            // Assert
+            action.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void Should_fail_with_descriptive_message_when_status_code_error_is_successful()
+        {
+            // Act
+            Action action = () =>
+                new HttpResponseMessage(HttpStatusCode.Gone).Should().BeSuccessful("because we want to test the failure {0}", "message");
+
+            // Assert
+            action
+                .Should().Throw<XunitException>()
+                .WithMessage("Expected HttpStatusCode to be successful (2xx) because we want to test the failure message, but found HttpStatusCode.Gone {value: 410}.");
+        }
```

Comment:
I know we're not consistent about this in the tests, but let's combine these two tests to a single one.
Verifying that `because` and `becauseArgs` are included in the failure message is part of the behavior when failing.

Reply:
But combining only these two tests makes it imho less consistent 🙂 wouldn't it be better to combine the other tests as well?

Reply:
Bad phrasing from my side.
Yes, let's combine each pair of `Throw<XunitException>()` and `Throw<XunitException>().WithMessage()` tests.